flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -1400,6 +1402,40 @@ mlir::Value IntrinsicLibrary::genMod(mlir::Type resultType,
   return genRuntimeCall("mod", resultType, args);
 }
 
+// MODULO
+mlir::Value IntrinsicLibrary::genModulo(mlir::Type resultType,
```

Comment:
Do you want to document the algorithm here?

Reply:
Sure, I added a comment describing what is the algorithm used for MODULO.